cli/azd/pkg/commands/pipeline/github_provider.go

```diff
@@ -358,17 +364,54 @@ func (p *GitHubCiProvider) configureConnection(
 	repoDetails *gitRepositoryDetails,
 	infraOptions provisioning.Options,
 	credentials json.RawMessage,
+	authType PipelineAuthType,
 	console input.Console) error {
 
 	repoSlug := repoDetails.owner + "/" + repoDetails.repoName
 	console.Message(ctx, fmt.Sprintf("Configuring repository %s.\n", repoSlug))
-	console.Message(ctx, "Setting AZURE_CREDENTIALS GitHub repo secret.\n")
 
 	ghCli := github.NewGitHubCli(ctx)
 	if err := p.ensureAuthorizedForRepoSecrets(ctx, ghCli, console, repoSlug); err != nil {
 		return fmt.Errorf("ensuring authorization: %w", err)
 	}
 
+	// Default to OIDC if not specified
+	if authType == "" {
+		authType = AuthTypeOidc
+	}
+
+	// OIDC + Terraform is not a valid combination - warn user.
+	if authType == AuthTypeOidc && infraOptions.Provider == provisioning.Terraform {
+		console.Message(ctx, output.WithWarningFormat("WARNING: Terraform provisioning does not support OIDC authentication, defaulting to Service Principal with client secret.\n"))
+		authType = AuthTypeClientSecret
+	}
+
+	var authErr error
+
+	switch authType {
+	case AuthTypeClientSecret:
+		authErr = p.configureClientSecretAuth(ctx, azdEnvironment, infraOptions, repoSlug, credentials, console)
+	default:
+		authErr = p.configureOidcAuth(ctx, azdEnvironment, infraOptions, repoSlug, credentials, console)
+	}
+
+	if authErr != nil {
+		return fmt.Errorf("failed configuring authentication: %w", authErr)
+	}
+
+	console.Message(ctx, fmt.Sprintf(
+		`GitHub Action secrets are now configured.
```

Comment:
What is our story to communicate to the user that the pipeline has to minimally contain

```yaml
permissions:
  id-token: write
  contents: read
```

defined?

Reply:
Documentation, I guess

Reply:
Added comment to official docs above this permissions block.